homeassistant/components/network/system_health.py

```diff
@@ -0,0 +1,59 @@
+"""Provide info to system health."""
+
+from typing import Any
+
+from homeassistant.components import system_health
+from homeassistant.core import HomeAssistant, callback
+
+from . import Adapter, async_get_adapters, async_get_announce_addresses
+from .models import IPv4ConfiguredAddress, IPv6ConfiguredAddress
+
+
+@callback
+def async_register(
+    hass: HomeAssistant, register: system_health.SystemHealthRegistration
+) -> None:
+    """Register system health callbacks."""
+    register.async_register_info(system_health_info, "/config/network")
+
+
+def _format_ips(ips: list[IPv4ConfiguredAddress] | list[IPv6ConfiguredAddress]) -> str:
+    return ", ".join([f"{ip['address']}/{ip['network_prefix']!s}" for ip in ips])
+
+
+def _get_adapter_ipv4_addresses(adapter: Adapter) -> str:
+    return f"{adapter['name']} ({_format_ips(adapter['ipv4'])})"
+
+
+def _get_adapter_ipv6_addresses(adapter: Adapter) -> str:
+    return f"{adapter['name']} ({_format_ips(adapter['ipv6'])})"
```

Comment:
These are very shallow methods, only used once. I would recommend just to inline it.

There is no real advantage in this case, besides adding additional jumps in code (which add to harder to follow code).